.gitlab/validate-agent-build/validate_agent_build.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+import sys
+import time
+
+import requests
+import os
+
+DATADOG_AGENT_PIPELINE_ID = 22
+BASE_URL = os.environ['CI_API_V4_URL']
+GITLAB_TOKEN = os.environ['GITLAB_TOKEN']
+STAGES_TO_CHECK = ['deps_fetch', 'source_test', 'binary_build', 'package_build']
+
+
+def _get_jobs(pipeline_id, scope=None):
+    all_jobs = []
+    url = f"{BASE_URL}/projects/22/pipelines/{pipeline_id}/jobs?per_page=20"
+    if scope is not None:
+        url += f"&scope={scope}"
+    while True:
+        resp = requests.get(url, headers={'PRIVATE-TOKEN': GITLAB_TOKEN})
```

Comment:
I think it's cleaner to do
```suggestion
    url = f"{BASE_URL}/projects/22/pipelines/{pipeline_id}/jobs"
    while True:
        resp = requests.get(url, headers={'PRIVATE-TOKEN': GITLAB_TOKEN}, params={"per_page": 20, "scope": scope})
```
If `scope` is `None` it will be ignored by `requests`.